### PR TITLE
Add docs section about executing coroutines 

### PR DESCRIPTION
**Pull request by juandaanieel**
Solves #360

### CODE REVIEW

**Comment by basnijholt** [comment]
Awesome @juandaanieel!

I just pushed a small change where I move the imports to the cell where they're used.

**Review by basnijholt — APPROVED** [review]
LGTM! I made a few small changes.

@akhmerov, merge when you're happy with it.

**Comment by akhmerov** [comment]
Yep! Want to go through the text with @juandaanieel, now it's rather minimal.

File: docs/source/tutorial/tutorial.advanced-topics.md [hunk 1]
```diff
@@ -367,6 +367,79 @@ await runner.task  # This is not needed in a notebook environment!
...
+Adaptive by itself does not implement a way of sharing partial results between function executions.
+Its implementation of parallel computation using executors is minimal by design.
+Instead the appropriate way to implement custom parallelization is by using coroutines (asynchronous functions).
+
+We illustrate this approach by using `dask.distributed` for parallel computations in part because it supports asynchronous operation out-of-the-box.
+Let us consider the computation below which has a slow but reusable part, and a fast part that cannot be reused.
+
+```{code-cell} ipython3
+import time
+
```
Comment (akhmerov) [inline docs/source/tutorial/tutorial.advanced-topics.md:381]: Please re-add the definition of `f` here: without it, the user won't know the goal and the computation is not fully defined.

File: docs/source/tutorial/tutorial.advanced-topics.md [hunk 2]
```diff
@@ -367,6 +367,79 @@ await runner.task  # This is not needed in a notebook environment!
...
+    return x**3
+
+
+def h(x):
+    """Fast function"""
+    return x**4
+```
+
+We need to convert `f` into a dask graph by using `dask.delayed`.
+In this example we will show how to send complex tasks to adaptive as coroutines.
```
Comment (akhmerov) [inline docs/source/tutorial/tutorial.advanced-topics.md:395]: Obsolete comment? Seems to repeat what we now say in the beginning.

**Review by akhmerov — CHANGES_REQUESTED** [review]

File: docs/source/tutorial/tutorial.advanced-topics.md [hunk 3]
```diff
@@ -367,6 +367,81 @@ await runner.task  # This is not needed in a notebook environment!
...
+
+@delayed
+def f(x, y):
+    return (x + y)**2
+```
+
+Next we define a computation using coroutines such that it reuses previously submitted tasks.
+
+```{code-cell} ipython3
+from collections import defaultdict
```
Comment (basnijholt) [inline docs/source/tutorial/tutorial.advanced-topics.md:413]: You do not seem to be using `defaultdict` anywhere.

File: docs/source/tutorial/tutorial.advanced-topics.md [hunk 4]
```diff
@@ -367,6 +367,81 @@ await runner.task  # This is not needed in a notebook environment!
...
+
+    future_f = client.compute(f(g_future, h(x % 1)))
+
+    return await future_f
+```
+
+Finally we provide the asynchronous function to the `learner` and run it via `AsyncRunner`.
+
+```{code-cell} ipython3
+learner = adaptive.Learner1D(async_h, bounds=(-3.5, 3.5))
```
Comment (basnijholt) [inline docs/source/tutorial/tutorial.advanced-topics.md:433]: You defined `f_parallel` but use an undefined `async_h`.

**Review by akhmerov — APPROVED** [review]

**Comment by akhmerov** [comment]
I reworded a couple of sentences. All good to go now, I think. Thanks again @juandaanieel!

**Pull request merged**